application/extractors.go

```diff
@@ -34,14 +31,6 @@ var URL Extractor = func(key string, req *Request) (interface{}, error) {
 		return nil, fmt.Errorf("URL %s is not valid", value)
 	}
 
-	mimetype := mime.TypeByExtension(filepath.Ext(value))
-
-	_, ok = image.Extensions[mimetype]
-
-	if !ok {
-		return nil, fmt.Errorf("Mimetype %s is not supported", mimetype)
-	}
```

Comment:
No problem. 

I can reintroduce the removed check, and if `mimetype` cannot be inferred by the extension first, then we can use [github.com/rubenfonseca/fastimage](github.com/rubenfonseca/fastimage) (`fastimage.DetectImageType(url)`) to detect the image type as a backup test.

This will cause a few bytes to be downloaded so it can inspect the file header, and will require a little remapping back to `image.Extensions`, but think this is worth it for an authoritative image type test that only comes into play if no extension is present.

@thoas what do you think?